docs/sql-ref-datatypes.md

```diff
@@ -706,3 +708,28 @@ The following table shows the type names as well as aliases used in Spark SQL pa
 </table>
 </div>
 </div>
+
+### Floating Point Special Values
+
+Spark SQL supports several special floating point values in a case-insensitive manner:
+
+ * Inf/+Inf/Infinity/+Infinity: positive infinity
+   * ```FloatType```: 1.0f / 0.0f
+   * ```DoubleType```: 1.0 / 0.0
+ * -Inf/-Infinity: negative infinity
+   * ```FloatType```: -1.0f / 0.0f
+   * ```DoubleType```: -1.0 / 0.0
+ * NaN: not a number
+   * ```FloatType```: 0.0f / 0.0f
+   * ```DoubleType```:  0.0d / 0.0
```

Comment:
Note: I've checked that the contents above are correct by https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L236-L251